beacon-chain/core/blocks/withdrawals_test.go

```diff
@@ -191,3 +196,385 @@ func TestProcessBLSToExecutionChange(t *testing.T) {
+					21: 0, 22: 0, 23: 0, 24: 0, 25: 0, 26: 0, 27: 0,
+				},
+			},
+		},
+		{
+			Args: args{
+				Name:                         "success more than max partially withdrawals",
+				NextWithdrawalIndex:          22,
+				LastWithdrawalValidatorIndex: 0,
+				PartialWithdrawalIndices:     []types.ValidatorIndex{1, 2, 3, 4, 5, 6, 7, 8, 9, 21, 22, 23, 24, 25, 26, 27, 29, 35, 89},
+				Withdrawals: []*enginev1.Withdrawal{
+					partialWithdrawal(1, 22), partialWithdrawal(2, 23), partialWithdrawal(3, 24),
+					partialWithdrawal(4, 25), partialWithdrawal(5, 26), partialWithdrawal(6, 27),
+					partialWithdrawal(7, 28), partialWithdrawal(8, 29), partialWithdrawal(9, 30),
+					partialWithdrawal(21, 31), partialWithdrawal(22, 32), partialWithdrawal(23, 33),
+					partialWithdrawal(24, 34), partialWithdrawal(25, 35), partialWithdrawal(26, 36),
+					partialWithdrawal(27, 37),
+				},
+			},
+			Control: control{
+				LastWithdrawalValidatorIndex: 27,
+				NextWithdrawalIndex:          38,
+				Balances: map[uint64]uint64{
+					1:  maxEffectiveBalance,
+					2:  maxEffectiveBalance,
+					3:  maxEffectiveBalance,
+					4:  maxEffectiveBalance,
+					5:  maxEffectiveBalance,
+					6:  maxEffectiveBalance,
+					7:  maxEffectiveBalance,
+					8:  maxEffectiveBalance,
+					9:  maxEffectiveBalance,
+					21: maxEffectiveBalance,
+					22: maxEffectiveBalance,
+					23: maxEffectiveBalance,
+					24: maxEffectiveBalance,
+					25: maxEffectiveBalance,
+					26: maxEffectiveBalance,
+					27: maxEffectiveBalance,
+				},
+			},
+		},
+		{
+			Args: args{
+				Name:                         "failure wrong number of partial withdrawal",
+				NextWithdrawalIndex:          21,
+				LastWithdrawalValidatorIndex: 37,
+				PartialWithdrawalIndices:     []types.ValidatorIndex{7},
+				Withdrawals: []*enginev1.Withdrawal{
+					partialWithdrawal(7, 21), partialWithdrawal(9, 22),
+				},
+			},
+			Control: control{
+				ExpectedError: true,
+			},
+		},
+		{
+			Args: args{
+				Name:                         "failure invalid withdrawal index",
+				NextWithdrawalIndex:          22,
+				LastWithdrawalValidatorIndex: 4,
+				FullWithdrawalIndices:        []types.ValidatorIndex{7, 19, 28, 1},
+				Withdrawals: []*enginev1.Withdrawal{
+					fullWithdrawal(7, 22), fullWithdrawal(19, 23), fullWithdrawal(28, 25),
+					fullWithdrawal(1, 25),
+				},
+			},
+			Control: control{
+				ExpectedError: true,
+			},
+		},
+		{
+			Args: args{
+				Name:                         "failure invalid validator index",
+				NextWithdrawalIndex:          22,
+				LastWithdrawalValidatorIndex: 4,
+				FullWithdrawalIndices:        []types.ValidatorIndex{7, 19, 28, 1},
+				Withdrawals: []*enginev1.Withdrawal{
+					fullWithdrawal(7, 22), fullWithdrawal(19, 23), fullWithdrawal(27, 24),
+					fullWithdrawal(1, 25),
+				},
+			},
+			Control: control{
+				ExpectedError: true,
+			},
+		},
+		{
+			Args: args{
+				Name:                         "failure invalid withdrawal amount",
```

Comment:
This test is the same as `"failure invalid validator index"`

Reply:
no it's not, the third withdrawal is a partial one